OsmAnd/src/net/osmand/plus/routepreparationmenu/WaypointsFragment.java

```diff
@@ -587,7 +587,7 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
 	}
 
 	private void deleteSwipeItem(int position) {
-		if (swipeDismissListener != null) {
+		if (swipeDismissListener != null && position < listView.getCount()) {
```

Comment:
looks like wrong fix

Reply:
reason of the issue:
https://github.com/osmandapp/Osmand/commit/bdf3ad58ef1ac6d5cf1f71014564d1d8d7953f37#diff-234d288dc83cf5d5e8ca2711f14e9b28R94

Reply:
ok, removed.